internal/server/httpServer.go

```diff
@@ -0,0 +1,90 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"server/internal/db"
+	util "server/pkg/util"
+)
+
+type HTTPServer struct {
+	httpServer *http.Server
+	DiceClient *db.DiceDB
+}
+
+// HandlerMux wraps ServeMux and forces REST paths to lowercase
+// and attaches a rate limiter with the handler
+type HandlerMux struct {
+	mux *http.ServeMux
+}
+
+func (cim *HandlerMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Convert the path to lowercase before passing to the underlying mux.
+	r.URL.Path = strings.ToLower(r.URL.Path)
+	cim.mux.ServeHTTP(w, r)
+}
+
+func NewHTTPServer(addr string, mux *http.ServeMux, client *db.DiceDB) *HTTPServer {
+	caseInsensitiveMux := &HandlerMux{
+		mux: mux,
+	}
+
+	return &HTTPServer{
+		httpServer: &http.Server{
+			Addr:              addr,
+			Handler:           caseInsensitiveMux,
+			ReadHeaderTimeout: 5 * time.Second,
+		},
+		DiceClient: client,
+	}
+}
+
+func (s *HTTPServer) Run(ctx context.Context) error {
+	var wg sync.WaitGroup
+
+	wg.Add(1)
```

Comment:
Just out of curiosity - why init a `sync.WaitGroup` and increment it but never call `.Wait` on it? @lucifercr07